### ISSUE TITLE
Docker Image with Pack

### ISSUE THREAD

**Issue by jamesward** [buildpacks/pack#468]
### Description
As part of the release process, create & publish a Docker image that contains pack, like:
https://github.com/GoogleCloudPlatform/cloud-builders-community/tree/master/pack

To use pack from systems like GitHub Actions, Cloud Build, etc it is useful to have an image with pack in it so that builder files can be written like:
https://github.com/jamesward/hello-php/blob/master/cloudbuild.yaml

**Comment by jromero** [comment]
Thanks for bringing this up. It makes sense.

Adding `state/discussion-needed` so that we can discuss this at the next working group meeting since it would require that we maintain another deliverable artifact. I don't anticipate any issues but would want to make sure everything is considered.

**Comment by thephw** [comment]
This approach is how at coca-cola we used Jenkins to run particular tools in a docker workspace without a node environment. Specifically in that instance we had a elasticsearch migration tool, [elastic-migrate](https://github.com/deftinc/elastic-migrate), written in node that was built as an executable docker image on release. For local development the image was used with a local docker volume mount to point to the migrations. Same setup for CI essentially. The project lifecycle we have for elastic-migrate [created several artifacts](https://github.com/deftinc/elastic-migrate/blob/master/bin/publish) GitHub releases, NPM releases, and dockerhub releases.

The big difference is that that CLI tool did not have docker as a dependency. @jamesward did you have thoughts on the approach to build an OCI image for `pack` when pack has docker as a dependency? Does that approach still achieve environment independence to be broadly applicable in other scenarios?

**Comment by hone** [comment]
@jamesward We talked about this in our WG meeting yesterday. This looks like an interesting use case, but there were some questions that came up:

* what platforms can run without extensive Docker-in-Docker support
* what behaviors would be enabled (create-builder, create-package, build, etc.)

Since this would add extra maintenance on one of the sub-teams and increase surface area, would you mind creating a RFC for it? Any of us on the team would be happy to help you with it.

**Comment by dgageot** [comment]
The classic way of running such image would be with the docker socket mounted. This would make it possible to run pack with a single `docker run` command on environments like Google Cloud Build, TravisCI and (I guess) GitHub Actions.

A Dockerfile for such image could be something as simple as:
```
FROM alpine:3.10

ARG PACK_VERSION
ENV PACK_URL https://github.com/buildpack/pack/releases/download/${PACK_VERSION}/pack-${PACK_VERSION}-linux.tgz
RUN wget -O- "${PACK_URL}" | tar -C /usr/local/bin -xz pack
```
(Copied from [Skaffold's codebase](https://github.com/GoogleContainerTools/skaffold/blob/master/deploy/buildpacks/Dockerfile))

or 

```
FROM busybox
ENV PACK_VERSION=v0.7.0
RUN wget -O- https://github.com/buildpack/pack/releases/download/${PACK_VERSION}/pack-${PACK_VERSION}-linux.tgz | tar zx

FROM gcr.io/distroless/base
ENTRYPOINT ["/pack"]
COPY --from=0 /pack /
```
(Copied from [GoogleCloudPlatform/cloud-builders-community](https://github.com/GoogleCloudPlatform/cloud-builders-community/blob/master/pack/Dockerfile))

**Comment by SocalNick** [comment]
[Asked about this in Slack])https://buildpacks.slack.com/archives/C94UJCNV6/p1581525852122900) and providing requested information:

* CI: Buildkite
* For images that need to interact with the VM docker daemon, we bind mount the Docker socket w/ `-v /var/run/docker.sock:/var/run/docker.sock`

**Comment by nebhale** [comment]
@SocalNick When you bind mount the Docker socket, do you typically have to do anything within the container to sanitize permissions?  And do you grant privileged status to the container?

**Comment by jromero** [comment]
Despite @nebhale's question which is valid and still worth answering, the @buildpacks/platform-maintainers have decided to move forward with this request. It will be prioritized accordingly.

If you are able to provide additional context into how it's used please feel free to add a comment to this issue. Insight is always welcomed.

**Comment by jromero** [comment]
Relevant conversation regarding dog fooding pack to create a pack image.

**TLDR; need to make improvements to launch operation to satisfy initial use case.**

https://buildpacks.slack.com/archives/CSFE49VGR/p1583001110014800?thread_ts=1583001110.014800&cid=CSFE49VGR

![Screenshot_20200301-140957](https://user-images.githubusercontent.com/475559/75632947-7ef72600-5bc6-11ea-9130-168ccaa2907f.jpg)

**Comment by dfreilich** [comment]
@jamesward I made an initial PR to create a Docker image; I assume you would want the entrypoint to be `pack`? I tried to build it with `pack`, but I found, as did @jromero, that there were some UX issues, given the current inability to set `ENTRYPOINTs` using `pack`.

**Comment by jamesward** [comment]
Nice! Yeah, setting the `ENTRYPOINT` to `pack` is ideal.  Thanks :)
